File: java/pe001.java
public class pe001 {
    public static void main(String[] args) {
        int ans = 0;
        for (int i = 3; i < 1000; i+=3)
        {
            ans += i;
        }
        for (int i = 5; i < 1000; i+=5)
        {
            ans += i;
        }
        for (int i = 15; i < 1000; i+=15)
        {
            ans -= i;
        }
        System.out.println(ans);
    }
}

File: rust/pe001.rs
//pe1
fn main() {
  let ans =
    (1..1000)
    .filter(|x| x % 3 == 0 || x % 5 == 0)
    .collect::<Vec<u32>>()
    .iter()
    .fold(0, |x, y| x + y);
  println!("{}", ans);
}